src/backend/internal/controller/chat/chat.go

```diff
@@ -334,11 +336,62 @@ func (c *Chat) ServeWS(gc *gin.Context) {
 
 	defer ws.Close()
 
-	user := gc.MustGet("user")
-	userModel := user.(model.User)
+	token := gc.Param("session-token")
+	if token == "" {
+		gc.JSON(gc.GetInt("errorStatus"), gin.H{
+			"status": "failed",
+			"error":  gc.GetString("errorMessage"),
+		})
+		gc.Abort()
+		return
+	}
```

Comment:
I think the `errorStatus` and `errorMessage` has not been set before arriving at this line of code. This will make the application panic and stop all operations.

I sugguest you may want to use the `util.Raise400Status(gc, "YOUR_DESIRED_MESSAGE")` instead.